src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java

```diff
@@ -304,10 +334,55 @@ private ResponseEntity<String> makeHarvestingByFile(FileSystemResource dataset,
         new HttpEntity<>(body, requestHeaders), String.class, "testDataset");
   }
 
-  private String removeCreationDate(String body){
+  private String removeCreationDate(String body) {
     JSONObject jsonObject = new JSONObject(body);
     jsonObject.getJSONObject("dataset-info").remove("creation-date");
     return jsonObject.toString();
   }
 
+
+  void assertJson(String expected, String actual) throws JsonProcessingException {
+    ObjectMapper objectMapper = new ObjectMapper();
+    assertTrue(objectMapper.readTree(expected).equals(objectMapper.readTree(actual)));
+  }
+
+  int extractDatasetId(String value) {
+    try {
+      ObjectMapper objectMapper = new ObjectMapper();
+      JsonNode node = objectMapper.readTree(value);
+      return node.get("dataset-id").asInt();
+    } catch (JsonProcessingException e) {
+      return -1;
+    }
+  }
+
+  String expectedDatasetInfoJson(String actual, String datasetId) {
+    ObjectMapper mapper = new ObjectMapper();
+    try {
+      JsonNode rootNode = mapper.readTree(actual);
+      JsonNode datasetNode = rootNode.get("dataset-info");
+      ((ObjectNode) datasetNode).remove("dataset-id");
+      ((ObjectNode) datasetNode).put("dataset-id", datasetId);
```

Comment:
The idea of updating this function was to use what was already in place and make it dynamic assertion. Because the test are running in parallel, with if test A returns dataset-id=1  and test B returns dataset-id=2, each test will have their own assertion during the test. So that is why was needed to replace the dataset-id on the tests. I am using the static content that was in place as a template to do an dynamic assertion.